packages/auth/src/auth.ts

```diff
@@ -0,0 +1,130 @@
+import PublicClientApplication, {
+    MSALAccount,
+    MSALResult,
+    MSALSilentParams,
+} from "react-native-msal";
+import type { MSALConfiguration } from "react-native-msal";
+import {
+    getMadAccount,
+    getMadAuthenticationResult,
+} from "./_internal/translation-layer/translationLayer";
+import { MadAccount, MadAuthenticationResult } from "./types";
+import { DEMO_USER, disableDemoMode, getIsDemoModeEnabled } from "./demo-mode";
+
+let pca: PublicClientApplication | null = null;
+
+async function _getInternalAccount(): Promise<MSALAccount | null> {
+    if (!pca) {
+        throw new Error("Unable to authenticate, pca is null");
+    }
+    const accounts: MSALAccount[] = await pca.getAccounts();
+
+    if (accounts.length > 0) {
+        const account = accounts[0];
+        return account;
+    }
+
+    return null;
+}
+
+async function _getMadAccountFromMSAL(): Promise<MadAccount | null> {
```

Comment:
It's all about semantics.

Undefined means it has not been defined (yet). Null means it has been defined to be nothing. Null is the correct datatype here because once we have tried to get the account, we either have an account, or nothing.

Look at `useAccount`. The value starts as `undefined`, cuz we simply haven't defined it yet. Then we try to get the account with `getAccount`. If we don't have an account, we can then set the value to `null`.